src/libtls/tls_server.c

```diff
@@ -464,17 +491,40 @@ static status_t process_client_hello(private_tls_server_t *this,
 		tls_named_group_t curve, requesting_curve = 0;
 		enumerator_t *enumerator;
 		chunk_t shared_secret = chunk_empty;
+		array_t *peer_key_shares;
+
+		/* parse peers key exchange key shares */
+		peer_key_shares = array_create(sizeof(key_share_t), 0);
+		extension = bio_reader_create(key_shares);
+		while (extension->remaining(extension))
+		{
+			key_share_t peer;
+
+			if (!extension->read_uint16(extension, &peer.curve) ||
+				!extension->read_data16(extension, &peer.key_share) ||
+				!peer.key_share.len)
+			{
+				DBG1(DBG_TLS, "invalid %N extension",
+					 tls_extension_names, extension_type);
+				this->alert->add(this->alert, TLS_FATAL, TLS_DECODE_ERROR);
+				return NEED_MORE;
+			}
+			array_insert_create_value(&peer_key_shares, sizeof(key_share_t),
+							 		  ARRAY_TAIL, &peer);
```

Comment:
```suggestion
									  ARRAY_TAIL, &peer);
```
But will look differently anyway if you use `array_insert()`.